File: hw2/main.go
package main

import (
	"fmt"
	"math"
)

func y(x float64) float64 {
	return math.Pow(math.Abs(math.Sin(math.Pow(x, 2)+1)), 1.0/3.0) - x
}

func main() {
	a, b := 0.0, 1.0
	e := 0.000001
	c := 0.0
	for ; math.Abs(b-a) > e; c = (a + b) / 2 {
		if y(a)*y(c) < 0 {
			b = c
		} else {
			a = c
		}
	}
	fmt.Println("Метод половинного деления Кавказов Алик")
	fmt.Println(c)
}

// Кавказов Алик БПО-21-01. Метод половинного деления

File: hw2/main.py
import math


def y(x):
    return (abs(math.sin(x**2+1)))**(1/3) - x


if __name__ == "__main__":
    a, b = 0, 1
    e = 0.000001
    c = (a+b)/2
    while True:
        if abs(b - a) < e:
            print(c)
            break
        print(y(a)*y(c))
        if y(a)*y(c) < 0:
            b = c
        else:
            a = c
        
        c = (a + b) / 2

     
    
